SIPS/sip-keeper-system.md

```diff
@@ -0,0 +1,165 @@
+
+## Rationale
+
+The rationale for a Central Keeper system contract is so that multiple contracts within the Synthetix platform requiring keepers can delegate the allocation of rewards and tracking to a single contract. Keepers also benefit from being able to track the reward payable from a single contract instead of checking all the Synthetix platform's contract for keeper functions.
+
+The Keeper system contract will be upgradeable to allow for fine-tuning and upgrading how the rewards will calculated for each keeper function. To enable this, the accumulated rewards of each keeper will be stored separetlely in eternal storage from the keeper contract implementation and the reward tokens held separately from the Synthetix keeper contract.
+
+Each keeper action will have an `allocation` of SNX / token rewards that will be allocated and topped up by the funding address or contract that provides the incentives for the keepr system. When the allocation for an action is depleted it will no longer be incentivising the particular action.
+
+## Technical Specification
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+
+### Synthetix Keeper System Contract
+
+The Keeper system contract will be upgradeable to allow for fine-tuning and upgrading how the rewards will calculated for each keeper function.
+
+A list of Synthetix internal contracts will have permission to notify the Keeper Incentives contract to record the keeper's address and what the performed function was to calculate and record the correct amount of token rewards.
+
+### Rollup Reward Claiming
+
+The SNX / token rewards keepers get for executing each keeper function will be recorded in the `Rewards Keeper contract` and a balance rolledup against their address for withdrawal.
+
+The total rewards can be withdrawn by the keeper at anytime instead of having the rewards transferred each time to reduce the gas costs of the actual operation. Keepers can also claim and withdraw their reward balance within the same transaction if they wish to utilise the reward as soon as possible.
+
+The gas costs of transferring the SNX rewards to the keeper per action can become prohiitively expensive and add to the costs of the keeper system.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
+Test cases for an implementation are mandatory for SIPs but can be included with the implementation.
+
+## Configurable Values (Via SCCP)
+
+Each action that a keeper could perform will have a configurable amount of USD value / SNX / token rewards that are paid to cover gas costs and running a keeper system.
+
+`gas: currency/eth rate` : 235.00 usd/eth
+
+| Action | Est Gas Cost | Gas Price | $USD Cost | $USD Reward |
+|--------|--------------|-----------|------------|------------|
+| 1. FreezeSynths for iSynths | 42,000| 35 gwei/gas | 0.34545 | 50.0 |
```

Comment:
I think this is too much detail for the SIP - the goal is to create a generalized keeper system, not to stipulate how much each should return.